Symfony/CS/Tests/FileCacheManagerIntegrationTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\SplFileInfo as FinderSplFileInfo;
+use Symfony\CS\FileCacheManager;
+use Symfony\CS\Fixer;
+
+class FileCacheManagerIntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    protected function setUp()
+    {
+        $this->fs = new Filesystem();
+        $this->rootDir = __DIR__.DIRECTORY_SEPARATOR.'Fixtures'.DIRECTORY_SEPARATOR.'FileCacheManagerIntegrationTest';
+
+        $this->fs->mkdir($this->rootDir);
+        $this->isCacheAvailable = $this->isCacheAvailable();
+
+        $this->clearFileCacheManagerIntegrationTestFiles();
+
+        $this->customConfigDir = $this->rootDir.DIRECTORY_SEPARATOR.'custom_config_dir';
+        $this->fileOne = $this->rootDir.DIRECTORY_SEPARATOR.'path_one'.DIRECTORY_SEPARATOR.'CommonFilename.php';
+        $this->fileTwo = $this->rootDir.DIRECTORY_SEPARATOR.'path_two'.DIRECTORY_SEPARATOR.'CommonFilename.php';
+
+        $this->fs->mkdir($this->rootDir);
+        $this->fs->mkdir($this->customConfigDir);
+        $this->fs->mkdir(dirname($this->fileOne));
+        $this->fs->mkdir(dirname($this->fileTwo));
```

Comment:
Ha, of course!

We dynamically declare a few fields here, but we never declare them. Personally, I would prefer to declare these fields:

```diff
diff --git a/Symfony/CS/Tests/FileCacheManagerIntegrationTest.php b/Symfony/CS/Tests/FileCacheManagerIntegrationTest.php
index 1040ca4..99aa69a 100644
--- a/Symfony/CS/Tests/FileCacheManagerIntegrationTest.php
+++ b/Symfony/CS/Tests/FileCacheManagerIntegrationTest.php
@@ -19,6 +19,31 @@

 class FileCacheManagerIntegrationTest extends \PHPUnit_Framework_TestCase
 {
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    /**
+     * @var string
+     */
+    private $rootDir;
+
+    /**
+     * @var bool
+     */
+    private $isCacheAvailable;
+
+    /**
+     * @var string
+     */
+    private $customConfigDir;
+
+    /**
+     * @var array
+     */
+    private $files;
+
     protected function setUp()
     {
         $this->fs = new Filesystem();
```

This has the advantage that you can

* search for usages of these fields (using PHPStorm)
* rename these fields (using PHPStorm refactor)
* avoid false negatives (using PHPStorm inspections)

Does that make sense?

As I said, not sure what the rules are in this repository. I have been working in other repositories where the maintainers had strong opinions on not declaring fields in tests.